drivers/include/sdp3x.h

```diff
@@ -0,0 +1,204 @@
+/*
+ * Copyright (C) 2019 Dirk Ehmen
+ *               2020 Nishchay Agrawal
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    drivers_sdp3x SDP3x temperature and differential pressure sensor
+ * @ingroup     drivers_sensors
+ * @{
+ * @file
+ * @brief       Device driver interface for the SDP3x sensor.
+ *
+ * @author      Dirk Ehmen       <ehmen@ibr.cs.tu-bs.de>
+ * @author      Nishchay Agrawal <f2016088@pilani.bits-pilani.ac.in>
+ * @}
+ */
+
+#ifndef SDP3X_H
+#define SDP3X_H
+
+#include "saul.h"
+#include "xtimer.h"
+#include "periph/i2c.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief   SDP3x models
+ */
+typedef enum {
+    SDP3x_31 = 1,
+    SDP3x_32 = 2
+} sdp3x_model_t;
+
+/**
+ * @brief   Device initialization parameters
+ */
+typedef struct {
+    i2c_t i2c_dev;                  /**< I2C device which is used */
+    uint8_t i2c_addr;               /**< I2C address */
+    sdp3x_model_t model;            /**< Model Number */
+    gpio_t irq_pin;                 /**< IRQ pin to check data ready */
+} sdp3x_params_t;
+
+/**
+ * @brief   Device descriptor for the SDP3x sensor
+ */
+typedef struct {
+    sdp3x_params_t params;    /**< Device initialization parameters */
+} sdp3x_t;
+
+/**
+ * @brief   Measurement from SDP3x sensor
+ */
+typedef struct {
+    double differential_pressure;  /**< Differential Pressure measured*/
+    double temperature;            /**< Temperature measured*/
```

Comment:
Please avoid using floating point math in drivers. Most MCUs don't have a FPU and even the larger ones (Cortex-M4F) will only support single precision `float`. 
No need to blow up the code by pulling in soft-float libraries - I'm pretty sure you can use decimal math here just fine.